### ISSUE TITLE
[Warlock] Demonic Pact 

### ISSUE THREAD

**Issue by ih8mondays** [dalaranwow/dalaran-wow#5465]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
The first time yout pet procs Demonic Pact you gain 10% extra spell power.
When your pet re-applies Demonic Pact you gain 10% extra of the 10% you just applied.
This continues until the server rounds it down to 0 resulting in a bit above 11% spellpower instead of 10%.

**Expected behaviour**: Tell us what should happen instead
Demonic Pact should apply and not include its own buff in the calculation.

**Steps to reproduce the problem**:

Video of bug: https://www.youtube.com/watch?v=VgzCSn4mjME
Spec used: https://wotlk.evowow.com/?talent#IZ0GchkAbihsgzAoE00Vb

1.
First 20 seconds i'm testing if my gear was the issue, but it jumps from 
155      ->        171      ->      173
155*10%= 15,5  |  16*10%) = 1,6
171-155 = 16     |  173-171 = 2
Now ofc the numbers are so small the server does a lot of rounding up which is why i put on my gear for the next part.

2.     
3345      ->            3680             ->          3714         ->         3717
3345*10% = 334,5    | 335*10%    = 33,5    | 34*10%      = 3,4
3680-3345 = 335      | 3714-3680 = 34       | 3717-3714 = 3

3. 
So as a conclusion it very much seems like it's increasing the spellpower it itself provides by 10% and in this case i got 37 sp more than intended. In a raid scenario it would be around 50 sp too much.

**Include proofs for this behaviour**
I'm only guessing this is not the intended behavior since you might as well change the tooltip to 11% and a bit.

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by ih8mondays** [comment]
As a side note it is clearly visible in the video that the 20 sec cooldown of Demonic Pact isn't working either.
Hint: it refreshes before 25 sec left.

**Comment by Rushor** [comment]
Ref: https://github.com/TrinityCore/TrinityCore/issues/9652
